src/Utilities/Compiler/Options/SymbolNamesOption.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Analyzer.Utilities.Extensions;
+using Analyzer.Utilities.PooledObjects;
+using Microsoft.CodeAnalysis;
+
+namespace Analyzer.Utilities
+{
+    internal sealed class SymbolNamesOption : IEquatable<SymbolNamesOption>
+    {
+        public static readonly SymbolNamesOption Empty = new SymbolNamesOption();
+
+        private readonly ImmutableHashSet<string> _names;
+        private readonly ImmutableHashSet<ISymbol> _symbols;
+
+        private SymbolNamesOption(ImmutableHashSet<string> names, ImmutableHashSet<ISymbol> symbols)
+        {
+            Debug.Assert(!names.IsEmpty || !symbols.IsEmpty);
+
+            _names = names;
+            _symbols = symbols;
+        }
+
+        private SymbolNamesOption()
+        {
+            _names = ImmutableHashSet<string>.Empty;
+            _symbols = ImmutableHashSet<ISymbol>.Empty;
+        }
+
+        public static SymbolNamesOption Create(ImmutableArray<string> symbolNames, Compilation compilation, string optionalPrefix)
+        {
+            if (symbolNames.IsEmpty)
+            {
+                return Empty;
+            }
+
+            var namesBuilder = PooledHashSet<string>.GetInstance();
+            var symbolsBuilder = PooledHashSet<ISymbol>.GetInstance();
+
+            foreach (var name in symbolNames)
+            {
+                if (name.Equals(".ctor", StringComparison.Ordinal) ||
```

Comment:
>.ctor [](start = 33, length = 5)

Do you care about static constructors as well?

Reply:
Added handling for static cctors, but I cannot test them for CA1303 as you cannot explicitly invoke it with string arguments.